### PR TITLE
APIMF-3096: catch java.lang.ArithmeticException when dividing by 0

### PR DESCRIPTION
**Pull request by arielmirra**
- APIMF-3096: catch division by 0 in JVM payload validation
- add general exception catch when calling validators

### CODE REVIEW

File: amf-client/jvm/src/test/scala/amf/client/validation/JvmPayloadValidationTest.scala [hunk 1]
```diff
@@ -41,4 +37,14 @@ class JvmPayloadValidationTest extends ClientPayloadValidationTest with NativeOp
       report.results.asSeq.head.message shouldBe "Regex defined in schema could not be processed"
     }
   }
+
+  test("Validation against a number with multipleOf 0 should throw error") {
```
Comment (tomsfernandez) [inline amf-client/jvm/src/test/scala/amf/client/validation/JvmPayloadValidationTest.scala]: should throw `violation`. It is more specific

File: amf-client/jvm/src/test/scala/amf/client/validation/JvmPayloadValidationTest.scala [hunk 2]
```diff
@@ -41,4 +37,14 @@ class JvmPayloadValidationTest extends ClientPayloadValidationTest with NativeOp
...
+    amf.Core.init().asFuture.flatMap { _ =>
+      amf.Core.registerPlugin(PayloadValidatorPlugin)
+      val shape     = new ScalarShape().withDataType(DataTypes.Number).withMultipleOf(0)
+      val validator = shape.payloadValidator("application/json").asOption.get
+      val report    = validator.syncValidate("application/json", "0")
```
Comment (tomsfernandez) [inline amf-client/jvm/src/test/scala/amf/client/validation/JvmPayloadValidationTest.scala]: I would add another example with a different number. 0/0 is a special (?) division. Or modify it.

File: amf-client/shared/src/test/scala/amf/validation/RamlBodyPayloadValidationTest.scala [hunk 3]
```diff
@@ -41,10 +40,6 @@ class RamlBodyPayloadValidationTest extends ApiShapePayloadValidationTest {
               "{\"in\": 22337203685477999090}",
               conforms = true,
               Option("application/json")),
-      Fixture("Invalid required pattern property",
```
Comment (tomsfernandez) [inline amf-client/shared/src/test/scala/amf/validation/RamlBodyPayloadValidationTest.scala]: Why was this test erased?

File: amf-webapi/jvm/src/main/scala/amf/plugins/document/webapi/validation/remote/JvmPayloadValidator.scala [hunk 4]
```diff
@@ -175,6 +164,9 @@ case class JvmReportValidationProcessor(override val profileName: ProfileName,
       case e: InvalidJsonValue =>
         Seq(invalidJsonValidation("Invalid json value was provided", element, e))
 
+      case e: ArithmeticException if e.getMessage == "Division undefined" =>
```
Comment (tomsfernandez) [inline amf-webapi/jvm/src/main/scala/amf/plugins/document/webapi/validation/remote/JvmPayloadValidator.scala]: What other types of messages are there?

**Review by tomsfernandez — APPROVED** [review]

File: amf-client/jvm/src/test/scala/amf/client/validation/JvmPayloadValidationTest.scala [hunk 5]
```diff
@@ -41,4 +37,14 @@ class JvmPayloadValidationTest extends ClientPayloadValidationTest with NativeOp
       report.results.asSeq.head.message shouldBe "Regex defined in schema could not be processed"
     }
   }
+
```
Comment (tomsfernandez) [inline amf-client/jvm/src/test/scala/amf/client/validation/JvmPayloadValidationTest.scala:40]: . (Dummy message so that i can request changes)

**Review by tomsfernandez — CHANGES_REQUESTED** [review]

File: amf-client/jvm/src/test/scala/amf/client/validation/JvmPayloadValidationTest.scala (hunk 2 above)
Comment (arielmirra) [reply amf-client/jvm/src/test/scala/amf/client/validation/JvmPayloadValidationTest.scala]: nice catch, added another line here to validate against a positive number

File: amf-client/shared/src/test/scala/amf/validation/RamlBodyPayloadValidationTest.scala (hunk 3 above)
Comment (arielmirra) [reply amf-client/shared/src/test/scala/amf/validation/RamlBodyPayloadValidationTest.scala]: yes, adding it back

File: amf-webapi/jvm/src/main/scala/amf/plugins/document/webapi/validation/remote/JvmPayloadValidator.scala (hunk 4 above)
Comment (arielmirra) [reply amf-webapi/jvm/src/main/scala/amf/plugins/document/webapi/validation/remote/JvmPayloadValidator.scala]: there are more cases that throw ArithmeticExceptions that don't relate to dividing by zero such as:
- Overflowing of a data type when exact result is requested e.g. `Math.addExact(int x, int y)`
- Rounding on an operation that yields an inexact result e.g. `BigDecimal.ROUND_UNNECESSARY`

I've avoided catching these cases, as I want only the division by 0 errors to report an exact violation message. These other cases should occur, will fall onto the general catch I've added.

Comment (tomsfernandez) [reply amf-webapi/jvm/src/main/scala/amf/plugins/document/webapi/validation/remote/JvmPayloadValidator.scala]: excelent

File: amf-webapi/jvm/src/main/scala/amf/plugins/document/webapi/validation/remote/JvmPayloadValidator.scala [hunk 6]
```diff
@@ -175,6 +164,9 @@ case class JvmReportValidationProcessor(override val profileName: ProfileName,
       case e: InvalidJsonValue =>
         Seq(invalidJsonValidation("Invalid json value was provided", element, e))
 
+      case e: ArithmeticException if e.getMessage == "Division undefined" || e.getMessage == "Division by zero" =>
+        Seq(invalidJsonValidation("Division by 0 is undefined", element, e))
```
Comment (tomsfernandez) [inline amf-webapi/jvm/src/main/scala/amf/plugins/document/webapi/validation/remote/JvmPayloadValidator.scala]: Yo le pondria algo así `Can't divide by 0` mas directo y simple.

**Review by tomsfernandez — DISMISSED** [review]

File: amf-webapi/jvm/src/main/scala/amf/plugins/document/webapi/validation/remote/JvmPayloadValidator.scala (hunk 6 above)
Comment (arielmirra) [reply amf-webapi/jvm/src/main/scala/amf/plugins/document/webapi/validation/remote/JvmPayloadValidator.scala]: true

**Review by tomsfernandez — APPROVED** [review]

**Pull request merged**